sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorReceiver.java

```diff
@@ -184,7 +188,9 @@ protected ReactorReceiver(AmqpConnection amqpConnection, String entityPath, Rece
 
     @Override
     public Flux<AmqpEndpointState> getEndpointStates() {
-        return endpointStates.distinct();
+        return endpointStates
+            .distinct()
```

Comment:
I believe it should be `distinctUntilChanged()`. `distinct` keeps a set, so if you see INACTIVE -> ACTIVE -> INACTIVE states, it'll only output INACTIVE and ACTIVE but not the last state.

Reply:
ah, I see, good point! will switch to `distinctUntilChanged` .